### PR TITLE
removed redundant exception handler from commands, correlation id from header

### PR DESCRIPTION
**Pull request by williexu**
Fixes: https://github.com/Azure/azure-cli/issues/4288

-removed some redundant code
-correlation is now taken from the response header instead of from within the response message

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [na ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [na ] Each command and parameter has a meaningful description.
- [na ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 1]
```diff
@@ -39,22 +39,17 @@ def handle_long_running_operation_exception(ex):
         ex,
         fault_type='failed-long-running-operation',
         summary='Unexpected client exception in {}.'.format(LongRunningOperation.__name__))
+
     message = getattr(ex, 'message', ex)
     error_message = 'Deployment failed.'
 
     try:
-        correlation_id = json.loads(ex.response.content.decode())['properties']['correlationId']
+        correlation_id = ex.response.headers['x-ms-correlation-request-id']
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/commands/arm.py]: This should probably be a `.get` so in the event the header doesn't have this, the CLI won't crash and burn.

**Comment by derekbekoe** [comment]
Correlation Ids are also used here - https://github.com/Azure/azure-cli/blob/master/src/azure-cli-core/azure/cli/core/commands/__init__.py#L223.

It's used for showing progress.
Does this impact that?

File: src/azure-cli-core/azure/cli/core/commands/arm.py (hunk 1 above)
Comment (williexu) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: Thats the point of the try statement?
I can do a `.get` instead and remove the try statement but it seems consistent with the rest of the method.

**Comment by williexu** [comment]
@derekbekoe it doesn't impact that part, which happens upon a keyboard interrupt and will skip the code where the handle_long_running_operation() is called.

**Pull request merged**